### PR TITLE
[Core] Clean up `Scheme`

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

This PR includes changes in the `scheme.h` file in the `kratos/solving_strategies/schemes/` directory. Remove legacy methods, replace `typedef` by `using`, remove legacy include, use `pragma once`

**🆕 Changelog**

- [Clean up `Scheme`](https://github.com/KratosMultiphysics/Kratos/commit/db881de7c8b7cbd98b75b316e738a65b39ebf318)
    - Replaced `typedef` with `using` for more modern, clear syntax. This was done for various types like `ClassType`, `TDataType`, `TSystemMatrixType`, `TSystemVectorType`, `LocalSystemMatrixType`, `LocalSystemVectorType`, `TDofType`, `DofsArrayType`, `ElementsArrayType`, and `ConditionsArrayType`.
    - Removed redundant comments and improved comment formatting.
    - Removed redundant check for `KRATOS_SCHEME` definition and included `#pragma once` instead to prevent multiple inclusions of the header file.
    - Removed the deprecated `InitializeNonLinearIteration` methods for `Element` and `Condition` that were marked as legacy.

### CODE REVIEW

**Review by philbucher — DISMISSED** [review]

File: applications/StructuralMechanicsApplication/custom_strategies/custom_schemes/residual_based_relaxation_scheme.hpp
Comment (AlejandroCornejo) [inline applications/StructuralMechanicsApplication/custom_strategies/custom_schemes/residual_based_relaxation_scheme.hpp]: watch?

File: applications/StructuralMechanicsApplication/custom_strategies/custom_schemes/residual_based_relaxation_scheme.hpp
Comment (AlejandroCornejo) [inline applications/StructuralMechanicsApplication/custom_strategies/custom_schemes/residual_based_relaxation_scheme.hpp]: why not removing commented code

**Review by rubenzorrilla — CHANGES_REQUESTED** [review]
Parallel utilities changes must be done in a separate PR.

**Comment by loumalouomega** [comment]
> Parallel utilities changes must be done in a separate PR.

I agree, this started like 20 minutes clean up of code

image

File: kratos/utilities/parallel_utilities.cpp
Comment (philbucher) [inline kratos/utilities/parallel_utilities.cpp]: I would have done this long ago, but there is no (easy) equivalent for this with Cxx

@RiccardoRossi and I discussed this a lot

#9985 will be the proper solution for this, not requiring the threadId at all

**Review by philbucher — CHANGES_REQUESTED** [review]

File: kratos/utilities/parallel_utilities.cpp
Comment (loumalouomega) [reply kratos/utilities/parallel_utilities.cpp]: True. In any case, right now current code that relies in OpenMPUtils will not work with SMP_CXX11, the code is more like a transition to remove OpenMPUtils. I can just simply revert change and wait until #9985 is merged

Comment (loumalouomega) [reply kratos/utilities/parallel_utilities.cpp]: Separated in https://github.com/KratosMultiphysics/Kratos/pull/11116

File: applications/StructuralMechanicsApplication/custom_strategies/custom_schemes/residual_based_relaxation_scheme.hpp
Comment (RiccardoRossi) [reply applications/StructuralMechanicsApplication/custom_strategies/custom_schemes/residual_based_relaxation_scheme.hpp]: this one is not portable...

**Review by RiccardoRossi — COMMENTED** [review]
changes look fine to me...apart obviously the ThreadId ...

**Comment by loumalouomega** [comment]
> changes look fine to me...apart obviously the ThreadId ...

I removed the thread id already

File: applications/StructuralMechanicsApplication/custom_strategies/custom_schemes/residual_based_relaxation_scheme.hpp
Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/custom_strategies/custom_schemes/residual_based_relaxation_scheme.hpp]: This is outdated, see message 

image

**Comment by RiccardoRossi** [comment]
yes, i saw... i am not approving since others were reviewing...but this is fine to me

**Review by philbucher — APPROVED** [review]

**Comment by loumalouomega** [comment]
@rubenzorrilla this is ready

**Review by rubenzorrilla — APPROVED** [review]
Only clean up. No behavior change.

**Pull request merged**
